src/components/find-on-shelf/FindOnShelfModal.tsx

```diff
@@ -24,25 +25,25 @@ import FindOnShelfPeriodicalDropdowns from "./FindOnShelfPeriodicalDropdowns";
 import { PeriodicalEdition } from "../material/periodical/helper";
 import { useConfig } from "../../core/utils/config";
 
-export const findOnShelfModalId = (faustId: FaustId) =>
-  `find-on-shelf-modal-${faustId}`;
+export const findOnShelfModalId = (faustIds: FaustId[]) => {
+  const sortedFaustIds = faustIds.sort();
+  return `find-on-shelf-modal-${sortedFaustIds.join("-")}`;
+};
```

Comment:
You using exactly same pattern earlier on.
I suggest crating a function that creates modalIds in a generic way:
```typescript
export const constructModalId = ({prefix, fragments}: {prefix: string, items: string[]}) =>
  `${prefix ? `${prefix}-` : ""}${items.join("-")}`;